8.2.1 Release

pookmish opened the pull request:
# DO NOT MERGE

pookmish committed:
Update drupal core to 8.8  and all contrib modules (#442)

* update dependencies and start eliminating config_installer

* cleaned up composer

* moved default content into submodule that gets enabled after profile installed

* fixup! moved default content into submodule that gets enabled after profile installed

* fixed unit test for views plugin

* replace old temp file path with new setting

* updated ds and its patch

* updated config after slick updated

* updated modules

bryanbraun committed:
feat(STN101,STN83):  node detail page layout and news in horizontal card pattern (#448)

* feat(STN-83): news content type with Colorful theme horizontal card pattern

- adds styles for colorful theme horizontal card to our ITCSS components/folder
- adds font weights and functions for colorful theme
- adds font mixins and font class utilities for Colorful theme headings and typography
- adds specific Colorful theme styles and class for horizontal `hb_cmp-horizontal-card` pattern
- adds Colorful theme variation function so you can find out the primary, secondary, and tertiary color for a certain variation, e.g., The colorful teal theme variation.
- also adds sass-true tests for this new function
- hides external link font awesome icon
- adds new URL field to horizontal card pattern (versus legacy pattern)
- prefixes all scss with `hb_` prefix
- changes all px to rem units
- alters some of the base styles

* feat(STN-101): add the three_column_w_image template & basic styles to humsci_basic

Co-authored-by: Lindsay R. Silver <lindsay@heysparkbox.com>

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

Commit:
fix: code climate scss-lint issues (#452)

- make all doublequote in scss single
- spacing and get rid of TODOs
- make all hb_ vars hb- for linting

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

pookmish committed:
Merge branch 'develop' into 8.2.1-release

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

kaseybon committed:
feat(88,105,106,109): Sparkbox Sprint 4 (#457)

* fix(STN-106): design chiseling on horizontal card

- refine heading sizes and resizing
- category hcard link style
- add image hover zoom
- add external link icon in background css

* feat(STN-105): vertical card

- Modified the vertical card pattern to match the horizontal card
- Renamed .hb-cmp-horizontal-card to .hb-cmp-card to allow the styles to be reused
- Applied .hb-cmp-card classes to the vertical card
- Added a .hb-cmp-card--horizontal modifier class to handle the horizontal card layout at larger breakpoints

* feat(STN-88): layout and styling for the news-item node detail page

- fix: remove commented code

* feat(STN-109) Add support for 2 and 3 column views (#458)

* feat(109): new grid patterns for 2 and 3 column pattern ui
* refactor(109): use calc for width of columns

Co-authored-by: Lindsay R. Silver <lindsay@heysparkbox.com>
Co-authored-by: Nathan Rambeck <nathan@rambeck.com>

* fixed phpunit coverage test from timing out

* refactor: update grid template and pattern yml

- use addClass instead of createClass in grid template
- add pattern examples

Co-authored-by: Lindsay R. Silver <lindsay@heysparkbox.com>
Co-authored-by: Bryan Braun <bbraun7@gmail.com>
Co-authored-by: Nathan Rambeck <nathan@rambeck.com>
Co-authored-by: pookmish <pookmish@stanford.edu>

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

pookmish committed:
Upgrade stanford media (#450)

* Updated stanford_media module

* Disabled modules via post update hook

* updated composer lock

* disable entity_browser during update process

* Fixed global and install behat tests

* fixed tests

* changed codecoverage filter

* Fixed hero overlay selectors

* HSD8-748 added missing diff.settings config

* changed circleci docker image (#459)

* changed circleci docker image

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

Commit:
feat: adds better preview for grid ui pattern (#460)

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

bryanbraun commented on config/default/core.entity_view_display.node.hs_news.hs_vertical_card.yml:
```diff
@@ -44,7 +44,7 @@ third_party_settings:
         settings:
           link: true
-          wrapper: span
+          wrapper: h2
```
@pookmish, I'm a little concerned about this one. You and I had talked about how "entity view display and entity form display configs for nodes are ignored," but I dug a little deeper and found this in the config settings:

https://github.com/SU-HSDO/suhumsci/blob/f2721bb8f82d0bbcc981c17ea2996a1479060db8/config/default/config_ignore.settings.yml#L4-L5

It looks to me like only the config for the _default_ view modes are ignored. This isn't a default view mode (it's a hs_vertical_card one). When I look on staging, **it looks like this updated the view mode on existing sites**, which wasn't what we were anticipating. Here's an example:

![image](https://user-images.githubusercontent.com/1256329/73405827-f2a1cc80-42c2-11ea-8872-9d954c8fe8d6.png)

I'm happy to change this back, but how would you feel about updating config-ignore to ignore these other view modes as well like so:

```diff
-  - 'core.entity_view_display.node.hs_*.default'
+  - 'core.entity_view_display.node.hs_*'
```

I think that would both solve this problem and give us more flexibility in customizing how UI patterns like vertical card work in our theme. Thoughts? Happy to discuss on a call if you want.

pookmish replied:
this would be fine with me. I didn't notice the change to the display was for a different view mode. I don't believe there would be any harm in ignoring all `hs_*` view displays. Feel free to submit a PR and we can verify the functionality there.
Good thing to notice

bryanbraun replied:
I have a PR up here: https://github.com/SU-HSDO/suhumsci/pull/461

I'm having some trouble testing it locally because I'm not familiar with the expected behavior for config ignore.

Does `drush cim` recognize the config changes but not apply them? Or is it supposed to not recognize changes made to those files?

Also, I'm trying to figure out how the `local/config_ignore.settings.yml` settings works alongside the settings in `config/default/config_ignore.settings.yml`. Do they both apply? Or just the local ones?

pookmish replied:
during a local `drush cim` the site will only ignore the `local`. All other environments ignore whats in `default`.
The PR looks fine to me. I don't foresee any issues with it.

bryanbraun committed:
fix: add all node view displays to config ignore (#461)

As a result:

- Updates to these displays will not impact existing sites, but will be
  applied as defaults to new sites created going forward.
- View mode configuration shouldn't be admin-locked in production allowing
  UI patterns with different fields (like those defined in other themes)
  to be chosen and configured.

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr449-g7xbltr9zlwut2wsbwfgd8z3psepmy8q.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4

Visual Diffs:
- https://dashboard.tugboat.qa/5e28849823b9ca6b4cf34cb4/visualdiffs

pookmish committed:
HSD8-749 Fixes for the media module & core updates (#464)

* HSD8-749 Fixed some media style issues and updated modules

* fixed video aspect ratio

* Fixed missing context for group blocks
